connectivity/drivers/cellular/CMakeLists.txt

```diff
@@ -1,10 +1,14 @@
-# Copyright (c) 2020 ARM Limited. All rights reserved.
+# Copyright (c) 2020-2021 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
 if("STMOD_CELLULAR" IN_LIST MBED_TARGET_LABELS)
     add_subdirectory(COMPONENT_STMOD_CELLULAR)
 endif()
 
+if("EP_ATLAS" IN_LIST MBED_TARGET_LABELS)
+    add_subdirectory(TARGET_EP_ATLAS/TELIT/ME310)
```

Comment:
What if another device has an `ME310`? Do we need to list them all here? Might it be better to use something like `DEVICE_HAS_ME310`?

Reply:
We provide ME310 generic driver. This inherits from it: `ONBOARD_TELIT_ME310 : public TELIT_ME310`  - from the file moved here, @ARMmbed/team-embeddedplanet Is this custom ME310 directly tailored for one board (on board module, not reusable) ?

`DEVICE_HAS`  is only for HAL (we kept it that way not to spread it anywhere else). connectivity/drivers use either COMPONENT or TARGET (one thing to clean up post old tools prime).

Reply:
This directory is for the ME310 onboard the EP_ATLAS. Others will use `connectivity/drivers/cellular/TELIT/ME310/`.

Reply:
My understanding was that a target that had an onboard modem needs to implement an `ONBOARD_<MODEM_NAME>` class to specialize the generic modem driver for that particular target (e.g., set target-specific config and pin usage - see [here](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/TARGET_EP_ATLAS/ONBOARD_TELIT_ME310.cpp#L160)). This way, the modem driver itself can be generic and shared across targets. I initially used the structure modeled by targets like [`WIO_BG96`](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L2128-L2153) which has an onboard Quectel BG96 - https://github.com/ARMmbed/mbed-os/tree/master/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_WIO_BG96.

Is there a more preferred way to handle this?

Reply:
> This way, the modem driver itself can be generic and shared across targets. I initially used the structure modeled by targets like WIO_BG96 which has an onboard Quectel BG96 - https://github.com/ARMmbed/mbed-os/tree/master/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_WIO_BG96.

One of the hidden gems, I'll create an issue about this, it will need to be fixed. For now, I would keep this file in the target and protect it with `MBED_CONF_CELLULAR_PRESENT` (I wonder why WIO has MBED_CONF_NSAPI_PRESENT - assuming if you got netsockets you got cellular 🙄 )

We need to address how to add/remove libraries within Mbed OS (from an app, you would do `target_link_libraries(mbed-os mbed-cellular)`. Let's keep this with `MBED_CONF_CELLULAR_PRESENT` so we can do proper clean up later.

I'll send PR fixing WIO_BG96 to use `MBED_CONF_CELLULAR_PRESENT` as well.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/36c34c59355a1c347c8c3d7c4533aa0f85bc44c6..811a72af88f5122024640db99af0a06607ea63e0).